source/slang/slang-ir-wgsl-legalize.cpp

```diff
@@ -1348,6 +1356,195 @@ struct LegalizeWGSLEntryPointContext
         }
     }
 
+    void hoistEntryPointParameterFromStruct(EntryPointInfo entryPoint)
```

Comment:
Are these exactly the same as metal? if so we can move it to a shared file and call the function from both backends.

Reply:
I initially copied them but I am getting crashes and test regressions, I will need more time to fix this. Another option is to just add new code to add semantic attributes to the parameters, but reusing code from Metal and refactoring(including refactoring of the existing) wgsl legalization code that is duplicated from Metal is still worth it IMO.